### ISSUE TITLE
PipeReader CopyToAsync an overload to copy To IBufferWriter<byte>

### ISSUE THREAD

**Issue by oliverjanik** [dotnet/runtime#2062]
At the moment (net core 3.1.1) `PipeReader.CopyToAsync` has 2 overloads: 
- copy to `Stream`
- copy to `PipeWriter` (which implements `IBufferWriter<byte>`)

https://docs.microsoft.com/en-us/dotnet/api/system.io.pipelines.pipereader.copytoasync

I'm trying to copy `PipeReader` into `ArrayBufferWriter<byte>` which seems like a common use-case and there's no direct way to do it. 

Of course I could use a `MemoryStream` but where's the fun in that?

**Comment by manandre** [comment]
In next release, a new `Stream.CopyToAsync` overload will be available as described in https://github.com/dotnet/corefx/issues/21472
```cs
public virtual Task CopyToAsync(Func<ReadOnlyMemory<byte>, object?, CancellationToken, ValueTask> callback, object? state, int bufferSize, CancellationToken cancellationToken)
```
Enabling this type of code:
```cs
await pipeReader
    .AsStream()
    .CopyToAsync(
        (memory, state, _) => ((IBufferWriter)state).Write(memory.Span),
        bufferWriter, 4096, default); 
```
Would it fit your need?

**Comment by oliverjanik** [comment]
Nah, I'm suggesting a convenience function. Your solution is longer than using ReadAsync in a while loop.

As a side note, is there any guidance on when to use streams vs the new pipelines?
